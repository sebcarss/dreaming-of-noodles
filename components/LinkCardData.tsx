export interface LinkCardData {
    country: string;
    region: string;
    gridLinkImage: string;
    gridLinkUrl: string;
}