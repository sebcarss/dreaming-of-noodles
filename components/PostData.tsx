export interface PostData {
    title: string;
    id: string;
    date: Date;
    tags: string[];
    published: boolean;
    excerpt: string;
    image: string;
};