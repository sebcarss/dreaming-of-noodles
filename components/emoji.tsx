type EmojiProps = {
    label?: string;
    symbol: string;
};

const Emoji = ({ label, symbol }: EmojiProps) => (
    <span
        data-testid="emoji"
        className="emoji"
        role="img"
        aria-label={label}
        aria-hidden={false}
    >
        {symbol}
    </span>
);

export default Emoji;