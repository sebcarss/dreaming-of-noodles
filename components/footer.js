export default function Footer() {
  return (
    <div>
      <footer className="row">
        <div className="col-12">Seb Carss</div>
      </footer>
    </div>
  );
}
