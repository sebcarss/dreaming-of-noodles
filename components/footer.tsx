export default function Footer() {
  return (
    <footer className="mb-3">
      <hr />
      <div className="lead text-center">Recipes by Seb Carss</div>
    </footer>
  );
}
