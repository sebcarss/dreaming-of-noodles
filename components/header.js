import Link from "next/link";

export default function Header({ home }) {
  const title = "Odyssey of Food"
  const subtitle = "Join my food exploration across the world"
  return (
    <header>
      {/* {home ? (
        <div className="text-center has-bg-img">
          <h1 className="display-4">{title}</h1>
          <h4 className="lead">{subtitle}</h4>
        </div>
      ) : (
        h2 used here for SEO purposes - could we just show the navbar instead?
        <h2 className="display-4">
          <Link href="/">
            <a>{title}</a>
          </Link>
        </h2>
      )} */}
    </header>
  );
}
