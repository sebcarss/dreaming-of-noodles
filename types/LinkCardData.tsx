export interface LinkCardData {
    frontmatter: { [key: string]: any };
    content: string;
}