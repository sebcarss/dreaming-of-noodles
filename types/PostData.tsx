export interface PostData {
    title: string;
    id: string;
    date: string;
    tags: string[];
    published: boolean;
    excerpt: string;
    image: string;
};