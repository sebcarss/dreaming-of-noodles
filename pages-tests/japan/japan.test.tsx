import Japan from "../../pages/japan";
import renderer from 'react-test-renderer';
import { LinkCardData } from "../../components/LinkCardData";
import { PostData } from "../../components/PostData";

const linkData: LinkCardData[] = [
  {
    country: "Japan",
    region: "Hokkaido",
    gridLinkImage: "japan-hokkaido.jpg",
    gridLinkUrl: "/japan/hokkaido",
  },
  {
    country: "Japan",
    region: "Aomori",
    gridLinkImage: "japan-aomori.jpg",
    gridLinkUrl: "/japan/aomori",
  },
  {
    country: "Japan",
    region: "Iwate",
    gridLinkImage: "japan-iwate.jpg",
    gridLinkUrl: "/japan/iwate",
  },
  {
    country: "Japan",
    region: "Miyagi",
    gridLinkImage: "japan-miyagi.jpg",
    gridLinkUrl: "/japan/miyagi",
  },
];

const allPostsData: PostData[] = [
  {
    title: "Hello World 2",
    id: "2019-01-02-hello-world-2",
    date: "2019-01-02",
    tags: ["hello", "world"],
    published: true,
    excerpt: "Hello excerpt",
    image: "https://source.unsplash.com/random/300x200",
  },
  {
    title: "Hello World 1",
    id: "2019-01-01-hello-world-1",
    date: "2019-01-01",
    tags: ["hello", "jam"],
    published: true,
    excerpt: "Hello 2 excerpt",
    image: "https://source.unsplash.com/random/300x200",
  },
];

describe("Japan page", () => {
  it('should match the snapshot', () => {
    const tree = renderer.create(<Japan posts={allPostsData} regionLinkData={linkData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
