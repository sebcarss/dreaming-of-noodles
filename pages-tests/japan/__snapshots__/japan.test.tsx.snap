// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Japan page should match the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <nav
      className="color-nav navbar navbar-expand-lg navbar-dark fixed-top"
      style={
        {
          "boxShadow": "0 10px 10px rgba(0,0,0,.2)",
          "color": "black",
          "fontSize": "1rem",
          "textAlign": "center",
          "width": "auto",
        }
      }
    >
      <a
        className="mx-2 navbar-brand"
        href="/"
      >
        Dreaming of Noodles
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="me-auto navbar-nav"
          onKeyDown={[Function]}
          style={
            {
              "textAlign": "left",
            }
          }
        >
          <a
            className="nav-link"
            data-rb-event-key="/japan"
            href="/japan"
            onClick={[Function]}
            onKeyDown={[Function]}
            style={
              {
                "color": "#db5004",
              }
            }
          >
            Japan
          </a>
          <a
            className="nav-link"
            data-rb-event-key="/recipe-index"
            href="/recipe-index"
            onClick={[Function]}
            onKeyDown={[Function]}
            style={
              {
                "color": "#db5004",
              }
            }
          >
            Recipe Index
          </a>
        </div>
      </div>
    </nav>
    <main
      className="mt-5"
    >
      <div
        className="mt-3 container"
      >
        <section>
          <div
            className="row"
          >
            <div
              className="col"
            >
              <div
                className="row"
              >
                <h1
                  className="text-left"
                >
                  Japan
                </h1>
                <hr />
              </div>
              <div
                className="row"
              >
                <p>
                  Japan is a country of diverse and rich cuisine, with each of its 47 prefectures offering a unique and delicious array of regional dishes. From Hokkaido in the north to Okinawa in the south, there is a world of flavors waiting to be explored.
                </p>
                <p>
                  Some of the most famous regional dishes include:
                </p>
                <p>
                  <strong>
                    Hokkaido
                  </strong>
                   - Known for its seafood, Hokkaido offers a variety of dishes featuring fresh ingredients from the sea, such as sushi, sashimi, and grilled fish.
                </p>
                <p>
                  <strong>
                    Kyoto
                  </strong>
                   - Famous for its traditional Japanese cuisine, Kyoto is home to Kaiseki, a multi-course meal featuring beautifully arranged dishes made with fresh, local ingredients.
                </p>
                <p>
                  <strong>
                    Osaka
                  </strong>
                   - The food capital of Japan, Osaka is famous for its street food and hearty dishes, such as Takoyaki (octopus balls), Okonomiyaki (savoury pancakes), and Kushikatsu (deep-fried skewered meat and vegetables).
                </p>
                <p>
                  <strong>
                    Hiroshima
                  </strong>
                   - A city in western Japan, Hiroshima is famous for its unique take on Okonomiyaki, where the ingredients are layered rather than mixed together.
                </p>
                <p>
                  With so many delicious regional dishes to explore, the culinary adventure in Japan never ends. Join us as we travel virtually through Japan, exploring the flavors and culture of each of its prefectures.
                </p>
              </div>
            </div>
          </div>
        </section>
        <div
          className="row"
        >
          <div>
            <div
              className="mt-3 row"
            >
              <h2
                style={
                  {
                    "textAlign": "center",
                  }
                }
              >
                Japanese Prefectures
              </h2>
            </div>
            <div
              className="g-1 mt-1 row row-cols-lg-6 row-cols-sm-4 row-cols-2"
            >
              <a
                href="/japan/hokkaido"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                <div
                  className="col"
                  style={
                    {
                      "backgroundColor": "white",
                      "borderRadius": "0.2em",
                      "boxShadow": "0 5px 10px rgba(0,0,0,.2)",
                      "color": "black",
                      "fontSize": "1rem",
                      "textAlign": "center",
                      "width": "auto",
                    }
                  }
                >
                  Hokkaido
                </div>
              </a>
              <a
                href="/japan/aomori"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                <div
                  className="col"
                  style={
                    {
                      "backgroundColor": "white",
                      "borderRadius": "0.2em",
                      "boxShadow": "0 5px 10px rgba(0,0,0,.2)",
                      "color": "black",
                      "fontSize": "1rem",
                      "textAlign": "center",
                      "width": "auto",
                    }
                  }
                >
                  Aomori
                </div>
              </a>
              <a
                href="/japan/iwate"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                <div
                  className="col"
                  style={
                    {
                      "backgroundColor": "white",
                      "borderRadius": "0.2em",
                      "boxShadow": "0 5px 10px rgba(0,0,0,.2)",
                      "color": "black",
                      "fontSize": "1rem",
                      "textAlign": "center",
                      "width": "auto",
                    }
                  }
                >
                  Iwate
                </div>
              </a>
              <a
                href="/japan/miyagi"
                onClick={[Function]}
                onMouseEnter={[Function]}
              >
                <div
                  className="col"
                  style={
                    {
                      "backgroundColor": "white",
                      "borderRadius": "0.2em",
                      "boxShadow": "0 5px 10px rgba(0,0,0,.2)",
                      "color": "black",
                      "fontSize": "1rem",
                      "textAlign": "center",
                      "width": "auto",
                    }
                  }
                >
                  Miyagi
                </div>
              </a>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <h2
            className="text-center mt-3"
          >
            Japanese Recipes
          </h2>
          <div
            className="col"
          >
            <div
              className="g-1 row row-cols-lg-3 row-cols-md-2 row-cols-1"
            >
              <div
                className="col"
              >
                <a
                  href="/posts/2019-01-02-hello-world-2"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <div
                    className="post-card h-100 card bg-dark text-white border-0"
                    style={
                      {
                        "borderRadius": 0,
                        "width": "auto",
                      }
                    }
                  >
                    <span
                      style={
                        {
                          "background": "none",
                          "border": 0,
                          "boxSizing": "border-box",
                          "display": "block",
                          "height": "initial",
                          "margin": 0,
                          "opacity": 1,
                          "overflow": "hidden",
                          "padding": 0,
                          "position": "relative",
                          "width": "initial",
                        }
                      }
                    >
                      <span
                        style={
                          {
                            "background": "none",
                            "border": 0,
                            "boxSizing": "border-box",
                            "display": "block",
                            "height": "initial",
                            "margin": 0,
                            "opacity": 1,
                            "padding": 0,
                            "paddingTop": "66.66666666666666%",
                            "width": "initial",
                          }
                        }
                      />
                      <img
                        alt="Hello World 2 - thumbnail"
                        data-nimg="responsive"
                        decoding="async"
                        onError={[Function]}
                        onLoad={[Function]}
                        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                        style={
                          {
                            "border": "none",
                            "bottom": 0,
                            "boxSizing": "border-box",
                            "display": "block",
                            "height": 0,
                            "left": 0,
                            "margin": "auto",
                            "maxHeight": "100%",
                            "maxWidth": "100%",
                            "minHeight": "100%",
                            "minWidth": "100%",
                            "objectFit": undefined,
                            "objectPosition": undefined,
                            "padding": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "width": 0,
                          }
                        }
                      />
                      <noscript>
                        <img
                          alt="Hello World 2 - thumbnail"
                          data-nimg="responsive"
                          decoding="async"
                          loading="lazy"
                          sizes="100vw"
                          src="/_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=3840&q=75"
                          srcSet="/_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=3840&q=75 3840w"
                          style={
                            {
                              "border": "none",
                              "bottom": 0,
                              "boxSizing": "border-box",
                              "display": "block",
                              "height": 0,
                              "left": 0,
                              "margin": "auto",
                              "maxHeight": "100%",
                              "maxWidth": "100%",
                              "minHeight": "100%",
                              "minWidth": "100%",
                              "objectFit": undefined,
                              "objectPosition": undefined,
                              "padding": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                              "width": 0,
                            }
                          }
                        />
                      </noscript>
                    </span>
                    <div
                      className="card-body"
                    >
                      <div
                        className="card-title h5"
                      >
                        Hello World 2
                      </div>
                      <p
                        className="text-muted card-text"
                      >
                        Hello excerpt
                      </p>
                    </div>
                  </div>
                </a>
              </div>
              <div
                className="col"
              >
                <a
                  href="/posts/2019-01-01-hello-world-1"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                >
                  <div
                    className="post-card h-100 card bg-dark text-white border-0"
                    style={
                      {
                        "borderRadius": 0,
                        "width": "auto",
                      }
                    }
                  >
                    <span
                      style={
                        {
                          "background": "none",
                          "border": 0,
                          "boxSizing": "border-box",
                          "display": "block",
                          "height": "initial",
                          "margin": 0,
                          "opacity": 1,
                          "overflow": "hidden",
                          "padding": 0,
                          "position": "relative",
                          "width": "initial",
                        }
                      }
                    >
                      <span
                        style={
                          {
                            "background": "none",
                            "border": 0,
                            "boxSizing": "border-box",
                            "display": "block",
                            "height": "initial",
                            "margin": 0,
                            "opacity": 1,
                            "padding": 0,
                            "paddingTop": "66.66666666666666%",
                            "width": "initial",
                          }
                        }
                      />
                      <img
                        alt="Hello World 1 - thumbnail"
                        data-nimg="responsive"
                        decoding="async"
                        onError={[Function]}
                        onLoad={[Function]}
                        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                        style={
                          {
                            "border": "none",
                            "bottom": 0,
                            "boxSizing": "border-box",
                            "display": "block",
                            "height": 0,
                            "left": 0,
                            "margin": "auto",
                            "maxHeight": "100%",
                            "maxWidth": "100%",
                            "minHeight": "100%",
                            "minWidth": "100%",
                            "objectFit": undefined,
                            "objectPosition": undefined,
                            "padding": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                            "width": 0,
                          }
                        }
                      />
                      <noscript>
                        <img
                          alt="Hello World 1 - thumbnail"
                          data-nimg="responsive"
                          decoding="async"
                          loading="lazy"
                          sizes="100vw"
                          src="/_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=3840&q=75"
                          srcSet="/_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fsource.unsplash.com%2Frandom%2F300x200&w=3840&q=75 3840w"
                          style={
                            {
                              "border": "none",
                              "bottom": 0,
                              "boxSizing": "border-box",
                              "display": "block",
                              "height": 0,
                              "left": 0,
                              "margin": "auto",
                              "maxHeight": "100%",
                              "maxWidth": "100%",
                              "minHeight": "100%",
                              "minWidth": "100%",
                              "objectFit": undefined,
                              "objectPosition": undefined,
                              "padding": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                              "width": 0,
                            }
                          }
                        />
                      </noscript>
                    </span>
                    <div
                      className="card-body"
                    >
                      <div
                        className="card-title h5"
                      >
                        Hello World 1
                      </div>
                      <p
                        className="text-muted card-text"
                      >
                        Hello 2 excerpt
                      </p>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
  <footer
    className="mb-3"
  >
    <hr />
    <div
      className="lead text-center"
    >
      Recipes by Seb Carss
    </div>
  </footer>
</div>
`;
